azure-devops/azext_devops/dev/common/utils.py

```diff
@@ -2,6 +2,35 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
+from knack.log import get_logger
+from knack.util import CLIError
+
+logger = get_logger(__name__)
+
+FILE_ENCODING_TYPES = ['ascii', 'utf-16be', 'utf-16le', 'utf-8']
+
+
+def read_file_content(file_path, encoding):
+    if not file_path or not encoding:
+        logger.debug("File path (%s) or encoding (%s) is missing.", file_path, encoding)
+        return None
+
+    if encoding in FILE_ENCODING_TYPES:
+        with open(file_path, 'r', encoding=encoding) as f:
+            try:
+                content = f.read()
+            except UnicodeDecodeError:
+                raise CLIError("Unable to decode file '{}' with '{}' encoding.".format(
```

Comment:
concern: we are eating the exception here.
maybe we can log it

Reply:
We are re-raising custom error. Not eating the exception.

Reply:
Will add debug trace.